packages/teams-js/src/private/externalAppCommands.ts

```diff
@@ -0,0 +1,192 @@
+import { sendMessageToParentAsync } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { ApiName, ApiVersionNumber, getApiVersionTag } from '../internal/telemetry';
+import { FrameContexts } from '../public';
+import { errorNotSupportedOnPlatform } from '../public/constants';
+import { runtime } from '../public/runtime';
+
+/**
+ * v2 APIs telemetry file: All of APIs in this capability file should send out API version v2 ONLY
+ */
+const externalAppCommandsTelemetryVersionNumber: ApiVersionNumber = ApiVersionNumber.V_2;
+
+/**
+ * @hidden
+ * Namespace to delegate the ActionCommand to the host
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @beta
+ */
+export namespace externalAppCommands {
+  /**
+   * @hidden
+   * The payload of IActionCommandResponse
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface IActionCommandResponse {
```

Comment:
>IActionCommandResponse [](http://example.com/codeflow?start=19&length=22)

Why are all the properties on this interface both able to be `undefined` and `null`? What is the sematic difference between when one is `undefined` and one is `null`?

Reply:
(this same question applies to all the properties in this file that can be `undefined` or `null`)

Reply:
Thanks for pointing this out, I updated `null` to `undefinded`